testutil/testrelayer/relays.go

```diff
@@ -1,40 +1,58 @@
 package testrelayer
 
 import (
+	"context"
+	"fmt"
+	"strings"
 	"testing"
 
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
 	"github.com/stretchr/testify/require"
 
+	"github.com/pokt-network/poktroll/pkg/crypto"
 	"github.com/pokt-network/poktroll/pkg/relayer"
+	testutilkeyring "github.com/pokt-network/poktroll/testutil/testkeyring"
 	servicetypes "github.com/pokt-network/poktroll/x/service/types"
 	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
 )
 
-// NewMinedRelay returns a new mined relay with the given session start and end
-// heights on the session header, and the bytes and hash fields populated.
-func NewMinedRelay(
+// NewUnsignedMinedRelay returns a new mined relay with the given session data,
+// as well as the bytes and the hash fields populated.
+//
+// It DOES NOT populate the signature fields and should only be used in contexts
+// where a partial mined relay is enough for testing purposes.
+//
+// TODO_IMPROVE: It does not (yet) verify against and adhere to the actual
+// relay mining difficulty of the service at hand.
```

Comment:
NB: For anywhere that this is required, we also have some pre-mined relays in pkg/relayer/miner/relay_fixtures_test.go that we should probably move to a testutil package (similar to pre-generated accounts). The generator for these pre-mined relay fixtures is parameterized. See: 
 https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/miner/relay_fixtures_test.go and https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/miner/gen.